library/manageiq_custom_attributes.py

```diff
@@ -0,0 +1,254 @@
+        """
+        added, updated = [], []
+        message = ""
+        # check if entity with the type and name passed exists in manageiq
+        entity_id = self.find_entity_by_name(entity_type, entity_name)
+        if entity_id:  # entity exists
+            ca_list = eval(custom_attributes)
+            entity_cas = self.get_entity_cas(entity_type, entity_id)
+            if entity_cas:
+                for new_ca in ca_list:
+                    existing_ca = next((ca for ca in entity_cas if ca['name'] == new_ca['name']), None)
+                    if existing_ca:
+                        if new_ca['value'] != existing_ca['value']:
+                            updated.extend(self.update_custom_attribute(entity_type, entity_id, new_ca, existing_ca['href']))
+                            message = "Successfuly updated the custom attributes of {entity_name} {entity_type}".format(entity_name=entity_name, entity_type=entity_type)
+                    else:
+                        added.extend(self.add_custom_attributes(entity_type, entity_id, [new_ca]))
+                        message = "Successfuly added the custom attributes to {entity_name} {entity_type}".format(entity_name=entity_name, entity_type=entity_type)
+            else:  # no custom attributes for this entity
+                added.extend(self.add_custom_attributes(entity_type, entity_id, custom_attributes))
+                message = "Successfuly added the custom attributes to {entity_name} {entity_type}".format(entity_name=entity_name, entity_type=entity_type)
+            if not added and not updated:
+                message = "The custom attributes are already added to {entity_name} {entity_type}".format(entity_name=entity_name, entity_type=entity_type)
+            return dict(
+                changed=self.changed,
+                msg=message,
+                updates={"Added": added, "Updated": updated}
+            )
+        else:  # entity doesn't exists
+            self.module.fail_json(msg="Failed to add the custom attributes. {entity_type} {entity_name} does not exist".format(entity_type=entity_type, entity_name=entity_name))
+
+    def delete_custom_attribute(self, ca, ca_href, entity_type, entity_id):
+        """ Returns the deleted custom attribute
+        """
+        try:
+            url = '{api_url}/{entity_type}/{id}/custom_attributes'.format(
+                    api_url=self.api_url,
+                    entity_type=ManageIQ.supported_entities[entity_type],
+                    id=entity_id)
+            ca_object = {'name': ca['name'], 'href': ca_href}
+            result = self.client.post(url, action='delete', resources=[ca_object])
+            self.changed = True
+            return result['results']
+        except Exception as e:
+            self.module.fail_json(msg="Failed to delete the custom attributes {ca}. Error: {error}".format(ca=ca, error=e))
+
+    def delete_custom_attributes(self, entity_type, entity_name, custom_attributes):
```

Comment:
This function has a lot of similarities with `add_or_update_custom_attributes`. Its worth the effort to see if it is possible to extract those similarities to a reusable function to simplify them both.